### ISSUE TITLE
Reduce Repo size by deploying via GitHub Actions

### ISSUE THREAD

**Issue by Skn0tt** [microsoft/playwright.dev#1461]
Currently, this site deploys to GitHub Actions via a Git Commit to a specific branch. This leads to a lot of heavyweight commits, and makes it very expensive to clone the repo. We can reduce this by using GitHub Actions to deploy the site.

**Comment by Skn0tt** [comment]
It looks like we're already doing this correctly, but we used a branch-based deploy in the past. I found [a nifty script](https://stackoverflow.com/a/66393128) that shows that this `gh-pages` branch indeed contributes >99.9% of the objects in the repository. Deleting it in GitHub should make those commits unreachable, and solve the problem.

The branch hasn't been commited to in 2 years, so I'm assuming it's fine to delete. If we need to recreate it: The current tip is at ff6f788b2451fb20351b3781dbdbe0b55a150e96. I'll go ahead and delete it.
